### DISCUSSION TITLE
Axis labels are getting clipped even after using `nice`

### DISCUSSION THREAD

**Discussion by shirshendubhowmick** [airbnb/visx#1188] [Q&A]
I am getting clipped axis babel for a specific time period even after using `nice: true`

![image](https://user-images.githubusercontent.com/12836504/116246986-14208b00-a788-11eb-96e5-ce6ef00500f4.png)

For different time periods the chart is getting rendered nicely

![image](https://user-images.githubusercontent.com/12836504/116247306-54800900-a788-11eb-8dbb-956f2fcbfe96.png)

Any suggestion on what might be wrong here ?

This is how my `timeScale` looks like
```TypeScript

const xMax = (props.parentWidth as number) - margin.left;

 scaleTime<number>({
   domain: [
     Math.min(...processedChartData.map(getDate)),
     Math.max(...processedChartData.map(getDate)),
   ],
   nice: true,
   range: [0, xMax],
 })
```

**Comment by williaster** [comment]
Hey @shirshendubhowmick 👋  thanks for checking out `visx`. I think your ticks are getting clipped because of the margin that you have set for your chart. `nice` has nothing to do with rendering, only the computation of tick values. From the [`d3-scale` docs](https://github.com/d3/d3-scale#continuous_nice):

> [nice] Extends the domain so that it starts and ends on nice round values.

Compared to HTML where you can make overflow visible, SVG content will get clipped if the container is not big enough (you can use dev tools to inspect where the bounds of the SVG are). So for your case I would either
- increase the right margin of the chart so it's not clipped, or
- right-justify the last tick using [`tickLabelProps`](https://airbnb.io/visx/docs/axis#Axis_tickLabelProps) to set `textAnchor: 'end'`

Hope that helps!
